library/alloc/src/ffi/c_str.rs

```diff
@@ -815,6 +815,24 @@ impl From<Vec<NonZero<u8>>> for CString {
     }
 }
 
+impl FromStr for CString {
+    type Err = NulError;
+
+    #[inline]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::new(s)
+    }
```

Comment:
It might be good to explicitly document that this does not require a terminating NUL in the input, maybe something like

```suggestion
    /// Creates a new C-compatible string from a `&str`.
    ///
    /// The trailing 0 byte will be appended by this function. If the provided data
    /// contains any 0 bytes in it, `Err(NulError)` will be returned.
    #[inline]
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        Self::new(s)
    }
```

or maybe just link to the docs for `CString::new`.

Reply:
Added a short description with a link to the corresponding function, and also did the same for the backward conversion.